lib/spacesuit/proxy_handler.ex

```diff
@@ -29,16 +29,17 @@ defmodule Spacesuit.ProxyHandler do
         stream(upstream, downstream)
 
       {:error, :econnrefused} ->
-        error_reply(req, 503, "Service Unavailable - Connection refused")
+        error_reply(req, 503, "SERVICE_UNAVAILABLE", "Connection refused")
 
       {:error, :closed} ->
-        error_reply(req, 502, "Bad Gateway - Connection closed")
+        error_reply(req, 502, "BAD_GATEWAY", "Connection closed")
 
       {:error, :timeout} ->
-        error_reply(req, 502, "Bad Gateway - Connection timeout")
+        error_reply(req, 502, "BAD_GATEWAY", "Connection timeout")
 
       {:error, :bad_request} ->
-        error_reply(req, 400, "Bad Request")
+        # What is this doing, the response for any bad request should be streamed through
+        error_reply(req, 400, "Bad Request", :nil)
```

Comment:
I have to read more about pattern matching in elixir... but the way it looks we're not passing through any response from the services in case of a BadRequest.
That response usually contains helpful details about why - for instance - json deserialization failed